### PR TITLE
[POAE7-2781] primitive type multiple case-when expr support in next codegen

### PR DESCRIPTION
**Pull request by haiweizh**
### What changes were proposed in this pull request?

[POAE7-2618] case/coalesce/if expr support in next codegen.
This story will be divided into four subtasks:

[POAE7-2675] primitive type if-then expr support in next codegen.
[POAE7-2781] primitive type multiple case-when expr support in next codegen.
[POAE7-2782] string type if/case expr support in next codegen.
[POAE7-2783] coalesce expr support in next codegen.
Implement [POAE7-2781] second base on [POAE7-2675].

### Why are the changes needed?

primitive type multiple case-when expr support in next codegen.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UTs.

### Which label does this PR belong to?

cider

### CODE REVIEW

File: src/cider/type/plan/Analyzer.cpp [hunk 1]
```diff
@@ -20,9 +20,12 @@
  * under the License.
  */
 #include "type/plan/Analyzer.h"
+#include "exec/nextgen/utils/JITExprValue.h"
 #include "exec/template/DateTimeUtils.h"
+#include "exec/template/IRCodegenUtils.h"
 #include "type/data/sqltypes.h"
```
Comment (kd97) [inline src/cider/type/plan/Analyzer.cpp:26]: seems no need to include `JITExprValue.h`, also  `IRCodegenUtils.h` is old code path, better to migrate/rename useful func?

File: src/cider/type/plan/CaseExpr.cpp [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+            *is_case = func.createLiteral(JITTypeTag::BOOL, true);
+          })
+          ->build();
+    }
+    func.createIfBuilder()
+        ->condition([&]() { return !is_case; })
+        ->ifTrue([&]() {
+          cider::exec::nextgen::utils::FixSizeJITExprValue else_jit_expr_value(
+              else_expr->codegen(context));
+          *value = *else_jit_expr_value.getValue();
```
Comment (jikunshang) [inline src/cider/type/plan/CaseExpr.cpp:254]: `value = else_jit_expr_value.getValue();` would be fine? `*` here will bring a memory copy I guess?

File: src/cider/type/plan/CaseExpr.cpp [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+          *null = *else_jit_expr_value.getNull();
+        })
+        ->build();
+    return set_expr_value(null, value);
+  } else if (case_ti.is_string()) {
+    UNIMPLEMENTED();
```
Comment (jikunshang) [inline src/cider/type/plan/CaseExpr.cpp:260]: add some log or todo

File: src/cider/type/plan/Analyzer.cpp (hunk 1 above)
Comment (haiweizh) [reply src/cider/type/plan/Analyzer.cpp:26]: Done.

File: src/cider/type/plan/CaseExpr.cpp (hunk 3 above)
Comment (haiweizh) [reply src/cider/type/plan/CaseExpr.cpp:260]: Done.

File: src/cider/type/plan/CaseExpr.cpp (hunk 2 above)
Comment (haiweizh) [reply src/cider/type/plan/CaseExpr.cpp:254]: Yes, this will do a memory copy. Use JITValuePointer.replace instant of * copy.

**Comment by haiweizh** [comment]
This pr will been closed. There is a pr with coalesce, plz visit [POAE7-2783].

**Pull request closed without merging**